### PR TITLE
人気Top3を商品詳細ページに追加しました

### PR DESCRIPTION
**Pull request by SekiMoemi**
確認お願いします。

### CODE REVIEW

File: src/main/java/jp/co/example/ecommerce_a/controller/ShowItemDetailController.java [hunk 1]
```diff
@@ -1,12 +1,16 @@
...
 
+import java.util.List;
+
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Controller;
 import org.springframework.ui.Model;
 import org.springframework.web.bind.annotation.GetMapping;
 import org.springframework.web.bind.annotation.RequestMapping;
 
+import jp.co.example.ecommerce_a.domain.Item;
```
Comment (SekiMoemi) [inline src/main/java/jp/co/example/ecommerce_a/controller/ShowItemDetailController.java:11]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.imports.CustomImportOrderCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Extra separation in import group before 'jp.co.example.ecommerce_a.domain.Item'

Comment (SekiMoemi) [inline src/main/java/jp/co/example/ecommerce_a/controller/ShowItemDetailController.java:11]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.imports.CustomImportOrderCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Wrong lexicographical order for 'jp.co.example.ecommerce_a.domain.Item' import. Should be before 'org.springframework.web.bind.annotation.RequestMapping'.

File: src/main/java/jp/co/example/ecommerce_a/controller/ShowItemDetailController.java [hunk 2]
```diff
@@ -1,12 +1,16 @@
...
 import jp.co.example.ecommerce_a.service.ShowItemDetailService;
+import jp.co.example.ecommerce_a.service.ShowItemListService;
```
Comment (SekiMoemi) [inline src/main/java/jp/co/example/ecommerce_a/controller/ShowItemDetailController.java:13]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.imports.CustomImportOrderCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Wrong lexicographical order for 'jp.co.example.ecommerce_a.service.ShowItemListService' import. Should be before 'org.springframework.web.bind.annotation.RequestMapping'.

File: src/main/java/jp/co/example/ecommerce_a/controller/ShowItemDetailController.java [hunk 3]
```diff
@@ -20,6 +24,8 @@ public class ShowItemDetailController {
 
 	@Autowired
 	private ShowItemDetailService showItemDetailService;
+	@Autowired
```
Comment (SekiMoemi) [inline src/main/java/jp/co/example/ecommerce_a/controller/ShowItemDetailController.java:27]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.FileTabCharacterCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Line contains a tab character.

Comment (SekiMoemi) [inline src/main/java/jp/co/example/ecommerce_a/controller/ShowItemDetailController.java:27]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.indentation.IndentationCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'member def modifier' has incorrect indentation level 8, expected level should be 2.

File: src/main/java/jp/co/example/ecommerce_a/controller/ShowItemDetailController.java [hunk 4]
```diff
@@ -20,6 +24,8 @@ public class ShowItemDetailController {
...
+	private ShowItemListService showItemListService;
```
Comment (SekiMoemi) [inline src/main/java/jp/co/example/ecommerce_a/controller/ShowItemDetailController.java:28]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.FileTabCharacterCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Line contains a tab character.

File: src/main/java/jp/co/example/ecommerce_a/controller/ShowItemDetailController.java [hunk 5]
```diff
@@ -30,7 +36,10 @@ public class ShowItemDetailController {
 	 */
 	@GetMapping("")
 	public String ShowItemDetail(String itemId, Model model) {
+		final int NUM_OF_SHOW_ITEM = 3;
```
Comment (SekiMoemi) [inline src/main/java/jp/co/example/ecommerce_a/controller/ShowItemDetailController.java:39]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.FileTabCharacterCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Line contains a tab character.

Comment (SekiMoemi) [inline src/main/java/jp/co/example/ecommerce_a/controller/ShowItemDetailController.java:39]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.indentation.IndentationCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'method def' child has incorrect indentation level 16, expected level should be 4.

Comment (SekiMoemi) [inline src/main/java/jp/co/example/ecommerce_a/controller/ShowItemDetailController.java:39]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.naming.AbbreviationAsWordInNameCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Abbreviation in name 'NUM_OF_SHOW_ITEM' must contain no more than '1' consecutive capital letters.

File: src/main/java/jp/co/example/ecommerce_a/controller/ShowItemDetailController.java [hunk 6]
```diff
@@ -30,7 +36,10 @@ public class ShowItemDetailController {
...
 		model.addAttribute("item", showItemDetailService.showItemDetail(Integer.parseInt(itemId)));
+		List<Item> itemList = showItemListService.searchByName("", "0").stream().limit(NUM_OF_SHOW_ITEM).toList();
```
Comment (SekiMoemi) [inline src/main/java/jp/co/example/ecommerce_a/controller/ShowItemDetailController.java:41]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.sizes.LineLengthCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Line is longer than 100 characters (found 122).

Comment (SekiMoemi) [inline src/main/java/jp/co/example/ecommerce_a/controller/ShowItemDetailController.java:41]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.FileTabCharacterCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Line contains a tab character.

Comment (SekiMoemi) [inline src/main/java/jp/co/example/ecommerce_a/controller/ShowItemDetailController.java:41]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.indentation.IndentationCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'method def' child has incorrect indentation level 16, expected level should be 4.

File: src/main/java/jp/co/example/ecommerce_a/controller/ShowItemDetailController.java [hunk 7]
```diff
@@ -30,7 +36,10 @@ public class ShowItemDetailController {
...
+		model.addAttribute("itemList", itemList);
```
Comment (SekiMoemi) [inline src/main/java/jp/co/example/ecommerce_a/controller/ShowItemDetailController.java:42]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.FileTabCharacterCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Line contains a tab character.

Comment (SekiMoemi) [inline src/main/java/jp/co/example/ecommerce_a/controller/ShowItemDetailController.java:42]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.indentation.IndentationCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'method def' child has incorrect indentation level 16, expected level should be 4.

File: src/main/java/jp/co/example/ecommerce_a/repository/ItemRepository.java [hunk 8]
```diff
@@ -90,7 +90,7 @@ public List<Item> findByTargetAgeGenruAndPriceRange(Integer targetAge, Integer g
 		return itemList;
 	}
 
-	 * 商品IDによってアイテム（おもちゃ）を取得するメソッド.
+	 /** 商品IDによってアイテム（おもちゃ）を取得するメソッド.
```
Comment (SekiMoemi) [inline src/main/java/jp/co/example/ecommerce_a/repository/ItemRepository.java:93]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.FileTabCharacterCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Line contains a tab character.

Comment (SekiMoemi) [inline src/main/java/jp/co/example/ecommerce_a/repository/ItemRepository.java:93]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.indentation.CommentsIndentationCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Block comment has incorrect indentation level 2, expected is 1, indentation should be the same level as line 99.

File: src/main/java/jp/co/example/ecommerce_a/service/ShowItemListService.java [hunk 9]
```diff
@@ -25,6 +25,8 @@ public class ShowItemListService {
 	/**
 	 * 名前でアイテム（おもちゃ）を曖昧検索する.
 	 * 
+	 * @param name    名前
```
Comment (SekiMoemi) [inline src/main/java/jp/co/example/ecommerce_a/service/ShowItemListService.java:28]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.javadoc.RequireEmptyLineBeforeBlockTagGroupCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Javadoc tag '@param' should be preceded with an empty line.

Comment (SekiMoemi) [inline src/main/java/jp/co/example/ecommerce_a/service/ShowItemListService.java:28]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.FileTabCharacterCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Line contains a tab character.

File: src/main/java/jp/co/example/ecommerce_a/service/ShowItemListService.java [hunk 10]
```diff
@@ -25,6 +25,8 @@ public class ShowItemListService {
...
+	 * @param sortNum ソートの条件
```
Comment (SekiMoemi) [inline src/main/java/jp/co/example/ecommerce_a/service/ShowItemListService.java:29]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.FileTabCharacterCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Line contains a tab character.

**Review by quangtung98 — APPROVED** [review]
問題がないと思いますので、承認します。

**Pull request merged**